lib/base-service.integration.spec.js

```diff
@@ -0,0 +1,540 @@
+              expect(error).toEqual(expectedError);
+              done();
+            });
+        });
+      });
+    });
+
+    describe('addToSet()', () => {
+      let clientAddToSetChildren, childrenToAddToSet, expectedChildren;
+
+      beforeEach(() => {
+        clientAddToSetChildren = Promise.promisify(client.addToSetChildren,
+          {'context': client});
+        const defaultChild = model1.children[0];
+        const childToBeAdded = {'name': 'Juan'};
+        childrenToAddToSet = [
+          defaultChild,
+          childToBeAdded,
+        ];
+        expectedChildren = [
+          defaultChild,
+          jasmine.objectContaining(childToBeAdded),
+        ];
+      });
+
+      it('should be add a sub-document', (done) => {
+        clientAddToSetChildren({'id': model1.id, 'children': childrenToAddToSet})
+          .then(() => {
+            return Model.findOne({'_id': model1.id});
+          })
+          .then((model) => {
+            const modelObject = CrudBaseService.getModelObject(model);
+            expect(modelObject.children).toEqual(expectedChildren);
+          })
+          .then(done);
+      });
+
+      describe('with invalid model id', () => {
+        it('should return model not found error', (done) => {
+          clientAddToSetChildren({'id': invalidId, 'children': childrenToAddToSet})
+            .catch((error) => {
+              expect(error).toEqual(expectedError);
+              done();
+            });
+        });
+      });
+    });
+
+    describe('remove()', () => {
+      let clientRemoveChildren, childToRemove, expectedChildren;
+
+      beforeEach(() => {
+        clientRemoveChildren = Promise.promisify(client.removeChildren,
+          {'context': client});
+        childToRemove = model1.children[0];
+        expectedChildren = [];
+      });
+
+      it('should remove the sub-documents', (done) => {
+        clientRemoveChildren({'id': model1.id, 'children': childToRemove})
+          .then(() => {
+            return Model.findOne({'_id': model1.id});
+          })
+          .then((model) => {
+            const modelObject = CrudBaseService.getModelObject(model);
+            expect(modelObject.children).toEqual(expectedChildren);
+          })
+          .then(done);
+      });
+
+      describe('with invalid model id', () => {
+        it('should return model not found error', (done) => {
+          clientRemoveChildren({'id': invalidId, 'children': childToRemove})
+            .catch((error) => {
+              expect(error).toEqual(expectedError);
+              done();
+            });
+        });
+      });
+
+      describe('with invalid child id', () => {
+        let invalidChild;
+
+        beforeEach(() => {
+          /* eslint-disable no-console */
+          console.warn = jasmine.createSpy('warn');
+          invalidChild = _.clone(childToRemove);
+          invalidChild.id = invalidId;
+        });
+
+        it('should return child not found error', (done) => {
```

Comment:
Cambiar descripción del it ya que no se prueba el retorno del error 'child not found error', en caso de regresar un error distinto al que se lanza con 'model not found error' agregar una prueba que exija específicamente ese error